src/api/spec/controllers/webui/packages/bs_requests_controller_spec.rb

```diff
@@ -1,19 +1,150 @@
+          it { expect(assigns[:bs_requests]).to include(outgoing_request) }
+          it { expect(assigns[:bs_requests]).to include(request_with_review) }
+        end
+      end
+
+      context 'when looking at the source_package requests' do
+        let(:source_project) { create(:project_with_package) }
+        let(:source_package) { source_project.packages.first }
+        let!(:incoming_request) do
+          create(:bs_request_with_submit_action,
+                 creator: user,
+                 priority: 'critical',
+                 source_package: create(:project),
+                 target_package: source_package)
+        end
+        let!(:outgoing_request) do
+          create(:bs_request_with_submit_action,
+                 creator: user,
+                 priority: 'critical',
+                 source_package: source_package,
+                 target_package: create(:project))
+        end
+        let!(:request_with_review) do
+          create(:bs_request_with_submit_action,
+                 review_by_package: source_package,
+                 creator: user,
+                 priority: 'critical')
+        end
+        let(:base_params) { { project: source_project, package: source_package, format: :json } }
+
+        before do
+          login user
+          Flipper.enable(:request_index, user)
+          get :index, params: params, format: :html
+        end
+
+        context 'and the direction parameters is "incoming"' do
+          let(:context_params) { { direction: 'incoming' } }
+
+          it { expect(response).to have_http_status(:success) }
+          it { expect(subject).to render_template(:index) }
+          it { expect(assigns[:bs_requests]).not_to include(outgoing_request) }
+          it { expect(assigns[:bs_requests]).to include(incoming_request) }
+          it { expect(assigns[:bs_requests]).not_to include(request_with_review) }
+        end
+
+        context 'and the direction parameters is "outgoing"' do
+          let(:context_params) { { direction: 'outgoing' } }
+
+          it { expect(response).to have_http_status(:success) }
+          it { expect(subject).to render_template(:index) }
+          it { expect(assigns[:bs_requests]).not_to include(incoming_request) }
+          it { expect(assigns[:bs_requests]).to include(outgoing_request) }
+          it { expect(assigns[:bs_requests]).not_to include(request_with_review) }
+        end
+
+        context 'and the direction parameters is "all"' do
+          let(:context_params) { { direction: 'all' } }
+
+          it { expect(response).to have_http_status(:success) }
+          it { expect(subject).to render_template(:index) }
+          it { expect(assigns[:bs_requests]).to include(incoming_request) }
+          it { expect(assigns[:bs_requests]).to include(outgoing_request) }
+          it { expect(assigns[:bs_requests]).to include(request_with_review) }
+        end
+      end
```

Comment:
Isn't it just a duplicated code of `context 'when looking at the target_package requests' do` ? It is swapping the `target_project` with `source_project` but the logic is pretty much the same. I'd rather drop one of the two. Am I missing something?

Reply:
Hm, let me take a look, maybe I can refactor it.